### PR TITLE
fix inline caching deoptimization

### PR DESCRIPTION
**Pull request by Uzlopak**
According to deoptigate, there are two inline caching issues.

Screenshot from 2021-11-28 13-42-07

This one fixes one of them.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by kibertoad** [comment]
@Uzlopak After https://github.com/fastify/busboy/pull/17 is merged, can you post benchmark results before and after this change?

File: deps/dicer/lib/Dicer.js [hunk 1]
```diff
@@ -175,9 +175,10 @@ Dicer.prototype._oninfo = function(isMatch, data, start, end) {
     this._part._read = function(n) {
       self._unpause();
     };
-    ev = this._isPreamble ? 'preamble' : 'part';
-    if (this._events[ev])
-      this.emit(ev, this._part);
+    if (this._isPreamble && this._events['preamble'])
```
Comment (kibertoad) [inline deps/dicer/lib/Dicer.js:178]: is it faster or slower, if there is one top-level if `if (this._isPreamble` and then another if about this._event inside of it?

**Comment by kibertoad** [comment]
@Uzlopak Can you pull latest master into this branch before running benchmark?

**Comment by Uzlopak** [comment]
Actually this benchmark is not representative. Will open an Issue.

File: deps/dicer/lib/Dicer.js (hunk 1 above)
Comment (Uzlopak) [reply deps/dicer/lib/Dicer.js:178]: isPreamble is always set to true or false. this._event[ev] could not exist.

**Comment by Uzlopak** [comment]
This PR does not have an effect on the performance currently. The hot path is still searchstream.

Screenshot from 2021-11-29 13-47-53

Probably if we optimize searchstream further more it could have an effect. :/

**Comment by kibertoad** [comment]
@Uzlopak Actually, I'm not sure we need all of this change, the way I read the report, it is unhappy due to ev being initialized as undefined and then changed into a string, and then being used as a key. Can you replace it just with a new const variable and see if that also addresses the cache deoptimization?

**Comment by kibertoad** [comment]
However, your approach still should be more efficient, as it replaces monomorphic variable with just a constant, which should be more predictable. Let's go with your implementation.

**Comment by kibertoad** [comment]
Can we drop `ev` from line 142 now? I think it should not be used anymore

**Comment by Uzlopak** [comment]
I removed the ev instantiation :)

**Review by kibertoad — APPROVED** [review]

**Pull request merged**